### ISSUE TITLE
How to apply to all views/viewsets?

### ISSUE THREAD

**Issue by marzique** [lingster/drf-api-tracking#46]
Is there any way to log all my api views without adding mixins to each of them?

**Comment by lingster** [comment]
you could create a derived class with the mixin added and make your views all derive from that new class? On 26 Feb 2021 16:25, Denys <notifications@github.com> wrote:
Is there any way to log all my api views without adding mixins to each of them?

**Comment by marzique** [comment]
Yeah, but even then I still need to put it manually to all of my views :(

**Comment by stardust85** [comment]
Maybe a middleware could easily include all views. It is extremely easy to write a middleware module for django (about 10 lines of  code for a middleware that does nothing but can access the request or response data).

**Comment by stardust85** [comment]
You can simply use https://github.com/django-request/django-request. As a bonus, it supports latest Django. Or at least it worked for me with Django 4.1 :)
